src/main/java/org/embulk/base/restclient/InputConfigDiffBuildable.java

```diff
@@ -0,0 +1,10 @@
+package org.embulk.base.restclient;
+
+import org.embulk.config.ConfigDiff;
+import org.embulk.spi.InputPlugin;
+import org.embulk.spi.Schema;
+
+public interface InputConfigDiffBuildable<T extends RestClientInputTaskBase>
+{
+    ConfigDiff buildInputConfigDiff(T task, Schema schema, int taskCount, InputPlugin.Control control);
```

Comment:
I think that `control` is not necessary. We need `List<TaskReport>` object that `control` returns instead.